shared/actions/chat.js

```diff
@@ -1011,22 +1012,28 @@ function _threadToPagination (thread) {
   }
 }
 
+// enforce no key collisions
+let timestampCounter = 1
 function _maybeAddTimestamp (message: Message, prevMessage: Message): MaybeTimestamp {
   if (prevMessage == null || prevMessage.type === 'Timestamp' || ['Timestamp', 'Deleted', 'Unhandled', 'Edit'].includes(message.type)) {
     return null
   }
   // messageID 1 is an unhandled placeholder. We want to add a timestamp before
   // the first message, as well as between any two messages with long duration.
   if (prevMessage.messageID === 1 || message.timestamp - prevMessage.timestamp > Constants.howLongBetweenTimestampsMs) {
+    timestampCounter++
     return {
       type: 'Timestamp',
       timestamp: message.timestamp,
-      key: message.timestamp,
+      key: `timestamp:${message.timestamp}:${timestampCounter}`,
```

Comment:
The timestamp dupe was being caused because `messageID === 1`. Since this can no longer be the case, this deduping logic wasn't needed.